lib/aclgenerator.py

```diff
@@ -271,48 +249,118 @@ class ACLGenerator(object):
 
   def __init__(self, pol, exp_info):
     """Initialise an ACLGenerator.  Store policy structure for processing."""
-    object.__init__(self)
-
-    # The default list of valid keyword tokens for generators
-    self._valid_keywords = self._REQUIRED_KEYWORDS.union(
-        self._OPTIONAL_SUPPORTED_KEYWORDS)
+    supported_tokens, supported_sub_tokens = self._getSupportedTokens()
 
     self.policy = pol
-
+    all_err = []
     for header, terms in pol.filters:
       if self._PLATFORM in header.platforms:
         # Verify valid keywords
         # error on unsupported optional keywords that could result
         # in dangerous or unexpected results
         for term in terms:
+          if term.platform:    
```

Comment:
this was removed in error.  I just popped it back in, and put it on top of the err = [] line so as to prevent that list creation for terms we don't need it on, eg they would be skipped from these checks.